lib/transaction.js

```diff
@@ -26,6 +30,19 @@ var Transaction = module.exports = function(sequelize, options) {
   } else {
     this.id = this.name = Utils.generateUUID();
   }
+
+  // `transaction.promise` implementation #2805
+  // main propose is to allow chainable logic inside hooks
+  this.promise = new Promise(function () {});
```

Comment:
Hmm yeah `promise._resolveFromResolver(Promise.reject() / Promise.resolve())` might be the best soluiton. Not exactly public API but still cleaner than saving a reference to the function ourselves :)